src/contract.rs

```diff
@@ -293,38 +311,38 @@ fn compute_current_metrics(
 
     let mut current_metrics: Vec<ValidatorMetrics> = vec![];
 
-    for update in previous_update_timings {
-        let ValidatorUpdateTimings {
-            addr: validator_addr,
-            ..
-        } = update;
+    for validator_addr in validators {
         let delegation_opt = deps
             .querier
             .query_delegation(&env.contract.address, validator_addr)?;
 
-        if delegation_opt.is_some() {
-            let delegation = delegation_opt.unwrap();
-            let current_rewards = get_total_rewards_in_vault_denom(
-                &delegation,
-                &vault_denom,
-                &mut exchange_rates_map,
-                &querier,
-            );
-
-            // This is the new Delegated amount after slashing Ex: (10 => 9.8 etc.,)
-            let current_delegated_amount = delegation.amount.amount.clone();
-            current_metrics.push(ValidatorMetrics {
-                addr: validator_addr.clone(),
-                rewards: current_rewards,
-                delegated_amount: current_delegated_amount,
-            });
-        } else {
-            // TODO: You should take a look at - Validator timings are already updated above
+        if delegation_opt.is_none() {
             return Err(ContractError::NoDelegationFound {
                 manager: env.contract.address.clone(),
                 validator: validator_addr.clone(),
             });
         }
+
+        let validator = deps.querier.query_validator(validator_addr)?.unwrap();
```

Comment:
You might want to add a check for unwrap here. 

I am not sure if validators show up here if suddenly they drop out of the validator set, either due to jailing or some other mishap.